src/util/drive.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::Ok;
+use google_drive3::hyper::client::HttpConnector;
+use google_drive3::hyper_rustls::HttpsConnector;
+use google_drive3::oauth2::ApplicationSecret;
+use google_drive3::{DriveHub, oauth2, hyper, hyper_rustls};
+
+use google_drive3::oauth2::authenticator_delegate::{DefaultInstalledFlowDelegate, InstalledFlowDelegate};
+use std::fs::File;
+use std::future::Future;
+use std::io::Write;
+use std::pin::Pin;
+use hyper::body::to_bytes;
+
+use crate::config ;
+
+async fn browser_user_url(url: &str, need_code: bool) -> Result<String, String> {
+    if webbrowser::open(url).is_ok() {
+        println!("webbrowser was successfully opened.");
```

Comment:
use "tracing::info!" instead of "println!"

Reply:
"def_delegate.present_user_url(url, need_code).await" returns "Result<String,String>, so it can't be applied "Result<String, anyhow::Error>"